pkg/e2e/infra/infra.go

```diff
@@ -123,40 +121,68 @@ func invalidMachinesetWithEmptyProviderConfig() *mapiv1beta1.MachineSet {
 	}
 }
 
+func buildMachineSetParams(client runtimeclient.Client, replicas int) e2e.MachineSetParams {
+	// Get the current workers MachineSets so we can copy a ProviderSpec
+	// from one to use with our new dedicated MachineSet.
+	workers, err := e2e.GetWorkerMachineSets(client)
+	o.Expect(err).ToNot(o.HaveOccurred())
+
+	providerSpec := workers[0].Spec.Template.Spec.ProviderSpec.DeepCopy()
+	clusterName := workers[0].Spec.Template.Labels[e2e.ClusterKey]
+
+	msName := e2e.RandomString()
```

Comment:
This needs rebase and follow https://github.com/openshift/cluster-api-actuator-pkg/pull/131/files#diff-798770aa2c88fce8a6789bb252bcb627R394